### PR TITLE
Play region preserves beat length across tempo changes

### PR DESCRIPTION
**Pull request by vsverchinsky**
Resolves: #5207

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

### CODE REVIEW

File: src/ProjectTempoListener.cpp [hunk 1]
```diff
@@ -53,4 +59,15 @@ void ProjectTempoListener::OnProjectTempoChange(double newTempo)
 {
    for (auto track : mTrackList)
       track->OnProjectTempoChange(newTempo);
+   
+   auto& viewInfo = ViewInfo::Get(mProject);
+   if(!viewInfo.playRegion.Empty() && mTempo > 0 && newTempo > 0)
```
Comment (saintmatthieu) [inline src/ProjectTempoListener.cpp:64]: Can `mTempo` possibly still be zero?

File: src/ProjectTempoListener.cpp [hunk 2]
```diff
@@ -27,12 +31,14 @@ static const AttachedProjectObjects::RegisteredFactory key {
 
 ProjectTempoListener::ProjectTempoListener(
    AudacityProject& project, TrackList& trackList)
-    : mTrackList { trackList }
+    : mProject { project }
+    , mTempo { ProjectTimeSignature::Get(project).GetTempo() }
```
Comment (saintmatthieu) [inline src/ProjectTempoListener.cpp:35]: Minor: To be consistent with the way `mTrackList` is obtained, I'd have opted for adding a `mViewInfo` member variable and called `ViewInfo::Get` from the outside.

**Review by saintmatthieu — APPROVED** [review]
NIce, thanks!

File: src/ProjectTempoListener.cpp (hunk 1 above)
Comment (vsverchinsky) [reply src/ProjectTempoListener.cpp:64]: I don't think that this is the possible case right now, but won't harm to check that. Non-positive tempo doesn't make sense, right?

File: src/ProjectTempoListener.cpp (hunk 2 above)
Comment (vsverchinsky) [reply src/ProjectTempoListener.cpp:37]: Good idea

File: src/ProjectTempoListener.cpp (hunk 1 above)
Comment (saintmatthieu) [reply src/ProjectTempoListener.cpp]: Sure, I just wanted to know if you were aware of a situation where it wasn't.

**Pull request merged**